### PR TITLE
Added code checkstyle

### PR DESCRIPTION
**Pull request by pdread100**
Using

mvn validate 

will cause the src to be style checked based on the config/checkstyle.xml. I purposefully turned it off from the normal build-release.sh since it will fail most of the code base. A number of projects seem to tie this to a git pre-commit action instead of the build action.

If this is not something that is found to be useful then I'll close this PR.

### CODE REVIEW

**Comment by pdread100** [comment]
The mvn install does not include the -Dcheckstyle.skip=-true, why would it, so I will hard wire a "skip" in the pom.xml and those who want to try it out will have to remove it.

**Comment by tnachen** [comment]
I think if you want to introduce a default style (which I think is great), we'll want to leave it on as default.
Looks like all the errors is valid style rules validations, can you fix them in the same PR? Otherwise I don't see why we add a code style but don't fix the errors.

**Comment by pdread100** [comment]
The intent was to get feedback first. Let folks test drive the style and
say yes or no. However I have no problem with fixing the errors and making
a PR if this style is ok.

File: config/checkstyle.xml [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+
+        <module name="WhitespaceAfter">
+            <!-- Checks that commas, semicolons and typecasts are followed by
+                 whitespace.
+            -->
+            <property name="tokens" value="COMMA, SEMI, TYPECAST"/>
+        </module>
+
+        <module name="NoWhitespaceAfter">
+            <!-- Checks that there is no whitespace after various unary operators.
```
Comment (tnachen) [inline config/checkstyle.xml]: Does this means it only valid to do:

true?1:2

As there is no whitespace allowed?

File: config/checkstyle.xml [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
+
+<!-- This is a checkstyle configuration file. For descriptions of
+what the following rules do, please see the checkstyle configuration
+page at http://checkstyle.sourceforge.net/config.html -->
+
+<module name="Checker">
+
+
+    <module name="FileTabCharacter">
```
Comment (tnachen) [inline config/checkstyle.xml]: +1

File: config/checkstyle.xml [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+        NAMING CHECKS
+
+        -->
+
+        <!-- Item 38 - Adhere to generally accepted naming conventions -->
+
+        <module name="PackageName">
+            <!-- Validates identifiers for package names against the
+              supplied expression. -->
+            <!-- Here the default checkstyle rule restricts package name parts to
```
Comment (tnachen) [inline config/checkstyle.xml]: I think this is too restrictive. We should remove this

**Comment by tnachen** [comment]
@erikdw @brndnmtthws What you guys think about this code style? IMO we should have one, just the question what styles we should put in. Personally modulo my comments, it seems ok to pull it in and modify it later when we don't like some of them.

**Comment by brndnmtthws** [comment]
It would be nice to standardize this with other projects. Maybe have a look at the Hadoop one:

https://github.com/mesos/hadoop/blob/master/checkstyle.xml

**Comment by pdread100** [comment]
I pulled this style from the mesos/myriad and made some minor mods to the
variable naming since it would not allow underscores at all. A lot of them
put in tabbing which gets pretty ugly.

File: config/checkstyle.xml (hunk 1 above)
Comment (pdread100) [inline config/checkstyle.xml]: Correct, true?1:2 would have to be

true ? 1 : 2

**Comment by tnachen** [comment]
I think this looks ok for now, we can change the style later if we like to

**Pull request merged**